pkg/planner/core/stats.go

```diff
@@ -275,6 +276,23 @@ func (ds *DataSource) initStats(colGroups [][]*expression.Column) {
 	ds.tableStats = tableStats
 	ds.tableStats.GroupNDVs = ds.getGroupNDVs(colGroups)
 	ds.TblColHists = ds.statisticTable.ID2UniqueID(ds.TblCols)
+	if config.GetGlobalConfig().Performance.LiteInitStats {
+		for _, col := range ds.tableInfo.Columns {
+			if col.State != model.StatePublic {
+				continue
+			}
+			// If we enable lite stats init, we need to register columns for async load.
+			_, isLoadNeeded, _ := ds.statisticTable.ColumnIsLoadNeeded(col.ID, false)
+			if isLoadNeeded {
+				statistics.HistogramNeededItems.Insert(model.TableItemID{
+					TableID:          ds.tableInfo.ID,
+					ID:               col.ID,
+					IsIndex:          false,
+					IsSyncLoadFailed: ds.SCtx().GetSessionVars().StmtCtx.StatsLoad.Timeout > 0,
+				}, false)
+			}
+		}
+	}
```

Comment:
Can we just move them into `CollectColumnStatsUsage()` to make it a unified place to handle this?

Reply:
Here we can also deal with the case that the SyncLoad is failed.